code/modules/antagonists/changeling/powers/summon_spiders.dm

```diff
@@ -1,18 +1,191 @@
+#define IDLE_AGGRESSIVE 0
+#define FOLLOW_AGGRESSIVE 1
+#define FOLLOW_RETALIATE 2
+#define IDLE_RETALIATE 3
+
 /datum/action/changeling/spiders
 	name = "Spread Infestation"
-	desc = "Our form divides, creating arachnids which will grow into deadly beasts."
-	helptext = "The spiders are thoughtless creatures, and may attack their creators when fully grown. Requires at least 5 stored DNA."
+	desc = "Our form divides, creating an aggressive arachnid which will regard us as a friend. Costs 45 chemicals."
+	helptext = "The spiders are thoughtless creatures, but will not attack their creators. Requires at least 7 stored DNA. Their orders can be changed via remote hivemind (Alt+Shift+Click)."
 	button_icon_state = "spread_infestation"
 	chemical_cost = 45
-	dna_cost = 1
-	req_dna = 5
+	dna_cost = 2
+	req_dna = 7
+	/// This var keeps track of the changeling's spider count
+	var/spider_counter = 0
+	/// Checks if changeling is already spawning a spider
+	var/is_operating = FALSE
 	power_type = CHANGELING_PURCHASABLE_POWER
 
-//Makes some spiderlings. Good for setting traps and causing general trouble.
+/// Makes a spider. Good for setting traps and combat.
 /datum/action/changeling/spiders/sting_action(mob/user)
-	for(var/i in 1 to 2)
-		var/obj/structure/spider/spiderling/S = new(user.loc)
-		S.grow_as = /mob/living/simple_animal/hostile/poison/giant_spider/hunter
+	if(is_operating) // To stop spawning multiple at once
+		return FALSE
+	is_operating = TRUE
+	if(spider_counter >= 3)
+		to_chat(user, "<span class='warning'>We cannot sustain more than three spiders!</span>")
+		is_operating = FALSE
+		return FALSE
+	user.visible_message("<span class='danger'>[user] begins vomiting an arachnid!</span>")
+	if(do_after(user, 4 SECONDS, FALSE, target = user)) // Takes 4 seconds to spawn a spider
+		spider_counter++
+		user.visible_message("<span class='danger'>[user] vomits up an arachnid!</span>")
+		var/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/S = new(user.loc)
+		S.owner_UID = user.UID()
+		S.faction |= list("spiders", "\ref[owner]") // Makes them friendly only to the owner & other spiders
+		SSblackbox.record_feedback("nested tally", "changeling_powers", 1, list("[name]"))
+		is_operating = FALSE
+		return TRUE
+	is_operating = FALSE
+	return FALSE
+
+/// Child of giant_spider because this should do everything the spider does and more
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider
+	/// References to the owner changeling
+	var/mob/owner_UID
+	/// To check if the spider died via gib or if it still needs to be gibbed, to prevent post-death revival skipping the limit
+	var/gibbed = FALSE
```

Comment:
See my other comment on `death`
```suggestion
```

Reply:
Had to re-do this due to death calling gib and gib calling death again therefor removing twice the counter.